src/perf/tracepoint/writer.cpp

```diff
@@ -20,31 +20,31 @@ Writer::Writer(int cpu, const EventFormat& event, trace::Trace& trace_,
   writer_(trace_.metric_writer((boost::format("tracepoint metrics for CPU %d") % cpu).str())),
   metric_instance_(
       trace_.metric_instance(metric_class, writer_.location(), trace_.system_tree_cpu_node(cpu))),
-  time_converter_(perf::time::Converter::instance())
+  time_converter_(perf::time::Converter::instance()),
+  metric_event_(otf2::chrono::genesis(), metric_class)
 {
-    counter_values_.resize(metric_class.size());
-    for (std::size_t i = 0; i < metric_class.size(); i++)
-    {
-        counter_values_[i].metric = metric_class[i];
-    }
 }
 
 bool Writer::handle(const Reader::RecordSampleType* sample)
 {
     auto tp = time_converter_(sample->time);
 
-    size_t index = 0;
-    for (const auto& field : event_.fields())
+    metric_event_.timestamp(tp);
+    for (const auto& index_field : nitro::lang::enumerate(event_.fields()))
```

Comment:
You can't use `nitro::lang::enumerate` here.
In case of: `field.is_integer()` the indices will be shifted. Not good.

Reply:
There even has been a commit in the past changing it from enumerate to this ordinary loop:
70d14732f4ea53c28e06d1856ba2e5ba4833fba3